hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -5011,10 +5016,34 @@ protected final OptimisticLockStyle optimisticLockStyle() {
 		return entityMetamodel.getOptimisticLockStyle();
 	}
 
+	@Override
 	public Object createProxy(Serializable id, SharedSessionContractImplementor session) throws HibernateException {
 		return entityMetamodel.getTuplizer().createProxy( id, session );
 	}
 
+	@Override
+	public Object createProxyForMappedSuperclass(
+			Serializable id, String mappedSuperclassName, SharedSessionContractImplementor session)
+			throws HibernateException {
+		return mappedSuperclassSubclassProxyFactory.get( mappedSuperclassName ).getProxy( id, session );
+	}
+
+	@Override
+	public void addMappedSuperclassSubclassProxyFacorty(String mappedSuperclassName, ProxyFactory proxyFactory) {
+		if ( mappedSuperclassSubclassProxyFactory == null ) {
+			mappedSuperclassSubclassProxyFactory = CollectionHelper.mapOfSize( 1 );
+		}
+		mappedSuperclassSubclassProxyFactory.put( mappedSuperclassName, proxyFactory );
+	}
+
+	@Override
+	public boolean isMappedSuperclassSubclass(String className) {
+		if ( mappedSuperclassSubclassProxyFactory == null ) {
+			return false;
+		}
+		return mappedSuperclassSubclassProxyFactory.containsKey( className );
+	}
```

Comment:
usually IDE would warn that the above code snippet can be simplified as
```
return mappedSuperclassSubclassProxyFactory != null && mappedSuperclassSubclassProxyFactory.containsKey( className );
```
Personally I agree with the IDE's suggestion. Ignore my comment if you disagree, though.